### PR TITLE
1953 grid list list tests

### PR DESCRIPTION
**Pull request by soturnn**
grid list and list tests added

### CODE REVIEW

**Comment by mf-code** [comment]
[5765] - Gz5CLQMB1LO8

**Comment by mf-code** [comment]
[5765] -

**Comment by mf-code** [comment]
[5765] - rCIhi4BYZSDY

**Comment by mf-code** [comment]
[5765] - FmbKROFJMUkt

**Comment by mf-code** [comment]
[5765] - yRRNrTePgE1H

**Comment by mf-code** [comment]
[5765] - IGEoTufkpIsY

**Comment by mf-code** [comment]
[5765] - VagIRHNJxqn3

**Comment by mf-code** [comment]
[5765] - ImNb3Z9bTIDA

**Comment by mf-code** [comment]
[5765] - vm5dMpwnDiBI

**Comment by mf-code** [comment]
[5765] - t87a0yw5TCDV

**Comment by mf-code** [comment]
[5765] - mOLoQw5hnN5K

**Comment by mf-code** [comment]
[5765] - 32qNy3LBJFYo

**Comment by mf-code** [comment]
[5765] - VOtAd3joJzhV

**Comment by mf-code** [comment]
[5765] - GpTdKoINyfoc

**Comment by mf-code** [comment]
[5765] - openjdk8 - Allure report on Netlify: https://5f08571515b410684d67cd0f--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5765] - openjdk10 - Allure report on Netlify: https://5f0858174d1eae024b337784--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5774] - MO3AqaF3s34K

**Comment by mf-code** [comment]
[5774] - UslzzGfEfVpx

**Comment by mf-code** [comment]
[5774] -

**Comment by mf-code** [comment]
[5774] - nVPlV52XrQ74

**Comment by mf-code** [comment]
[5774] - qWmfoKga9AvG

**Comment by mf-code** [comment]
[5774] - QD7cTkezfGVy

**Comment by mf-code** [comment]
[5774] - FG8myw0yfi29

**Comment by mf-code** [comment]
[5774] - VsM3olOjQL1q

**Comment by mf-code** [comment]
[5774] - r70yfnbVrdQk

**Comment by mf-code** [comment]
[5774] -

**Comment by mf-code** [comment]
[5774] -

**Comment by mf-code** [comment]
[5774] - JRs096RLGx9P

**Comment by mf-code** [comment]
[5774] - pmxVvrhT9Mk6

**Comment by mf-code** [comment]
[5774] - jzdg7ciRIqjy

**Comment by mf-code** [comment]
[5774] - openjdk8 - Allure report on Netlify: https://5f0c23cf84abef62fa56f9a0--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5774] - openjdk10 - Allure report on Netlify: https://5f0c24b645f6a6cfc14eea6c--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5791] - kqF9as5QApq1

**Comment by mf-code** [comment]
[5791] - Dulls9NFZATv

**Comment by mf-code** [comment]
[5791] - KvLe5nUByCyp

**Comment by mf-code** [comment]
[5791] - AF9Ktv3StV1X

**Comment by mf-code** [comment]
[5791] - eVksz4txwl9i

**Comment by mf-code** [comment]
[5791] - bQXpmKH6RbUd

**Comment by mf-code** [comment]
[5791] - C73DKw93I5Eu

**Comment by mf-code** [comment]
[5791] - xRnl2PJdQzxA

**Comment by mf-code** [comment]
[5791] - qneEHODTSCcA

**Comment by mf-code** [comment]
[5791] - yd6R4ZyqjJAi

**Comment by mf-code** [comment]
[5791] - X1qjLvQAMUrt

**Comment by mf-code** [comment]
[5791] - hNbnW1MXSyxa

**Comment by mf-code** [comment]
[5791] - XP30JnnLfRvr

**Comment by mf-code** [comment]
[5791] - xNYNpBTtl0eO

**Comment by mf-code** [comment]
[5791] - openjdk8 - Allure report on Netlify: https://5f0ddf96974d92578726fd5f--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5791] - openjdk10 - Allure report on Netlify: https://5f0de05dddc5f15a4894c5e4--musing-swartz-6fc060.netlify.app

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/GridListTests.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Test
+    public void basicGridListTextTest() {
+        gridListSection.basicGridList.get(1).is().text("1");
+    }
+
+    @Test
+    public void basicGridListColorTest() {
+        gridListSection.dynamicGridList.get(1)
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/GridListTests.java:32]: It looks like a typo: gridListSection.dynamicGridList. Probably meant gridListSection.bascisGridList?

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/GridListTests.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    @BeforeMethod
+    public void before() {
+        shouldBeLoggedIn();
+        angularPage.shouldBeOpened();
+    }
+
+    @Test
+    public void basicGridListBasicTest() {
+        gridListSection.basicGridList.is().displayed();
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/GridListTests.java:21]: Can you explain the logic of the test?
Usually, first preparation then verification of the result, but here it seems to me the opposite.

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/GridListTests.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    @Test
+    public void basicGridListColorTest() {
+        gridListSection.dynamicGridList.get(1)
+                 .has().css("background-color", "rgba(" + 173 + ", " + 216 + ", " + 230 + ", 1)");
+    }
+
+    @Test
+    public void dynamicGridListBasicTest() {
+        gridListSection.dynamicGridList.is().displayed();
+        gridListSection.dynamicGridList.get(1).show();
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/GridListTests.java:39]: The same about test logic?

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/GridListTests.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Test
+    public void dynamicGridListTextTest() {
+        gridListSection.dynamicGridList.get(1).is().text("One");
+    }
+
+    @Test
+    public void dynamicGridListColorTest() {
+        gridListSection.dynamicGridList.get(1)
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/GridListTests.java:49]: There are 4 different background colors. Maybe you should check them all out?

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    @BeforeMethod
+    public void before() {
+        shouldBeLoggedIn();
+        angularPage.shouldBeOpened();
+    }
+
+    @Test
+    public void basicListBasicTest() {
+        listSection.basicList.is().displayed();
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java:21]: The same about test logic?

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    @Test
+    public void basicListTextTest() {
+        listSection.basicList.get(1).is().text("Item 1");
+        listSection.basicList.get(2).is().text("Item 2");
+        listSection.basicList.get(3).is().text("Item 3");
+    }
+
+    @Test
+    public void listWithSectionsBasicTest() {
+        listSection.listWithSection.is().displayed();
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java:34]: The same about test logic?

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        listSection.listWithSection.get(1).show();
+    }
+
+    @Test
+    public void listWithSectionsTextTest() {
+        listSection.listWithSection.get(1).is().text("folder\nPhotos\nJan 1, 2016");
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java:40]: There are 5 different list sections. Maybe you should check them all out?

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+import io.github.epam.TestsInit;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static io.github.com.StaticSite.angularPage;
+import static io.github.com.pages.AngularPage.listSection;
+import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+
+public class ListTests extends TestsInit {
+
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java:12]: Probably also worth thinking about checking the icons?

**Comment by mf-code** [comment]
[5809] - aYzgDMVjHz2I

**Comment by mf-code** [comment]
[5809] - xZRGb06D9VIk

**Comment by mf-code** [comment]
[5809] - g9NLfr3hw6ys

**Comment by mf-code** [comment]
[5809] - MWt6T7zpzl2i

**Comment by mf-code** [comment]
[5809] - j96UGAwxLRMc

**Comment by mf-code** [comment]
[5809] - 9i7NDfw2qYOL

**Comment by mf-code** [comment]
[5809] - QZ5CaRMImJHz

**Comment by mf-code** [comment]
[5809] - jkItSmnqSl87

**Comment by mf-code** [comment]
[5809] - AfRc3wxeF0Ez

**Comment by mf-code** [comment]
[5809] - wY2OYOCLOSSG

**Comment by mf-code** [comment]
[5809] - JTAB44M76ICs

**Comment by mf-code** [comment]
[5809] - DoXJmgq3aBqH

**Comment by mf-code** [comment]
[5809] - mo8yaA0bQ6m2

**Comment by mf-code** [comment]
[5809] - n2dIomYGL00D

**Comment by mf-code** [comment]
[5809] - openjdk8 - Allure report on Netlify: https://5f0f643258f63201f087641c--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5809] - openjdk10 - Allure report on Netlify: https://5f0f64d71880730199e7a4f2--musing-swartz-6fc060.netlify.app

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+import io.github.epam.TestsInit;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static io.github.com.StaticSite.angularPage;
+import static io.github.com.pages.AngularPage.gridListSection;
+import static io.github.com.pages.AngularPage.listSection;
+import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+import static org.hamcrest.core.StringContains.containsString;
+
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java:12]: There are many unused import statements in the class.
Please format file (delete unused import statements).

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java (hunk 8 above)
Comment (soturnn) [reply jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ListTests.java:14]: @gbuh , all edits made

**Comment by mf-code** [comment]
[5810] - c72sU4rc7bzO

**Comment by mf-code** [comment]
[5810] - vU2E6YVzE3KY

**Comment by mf-code** [comment]
[5810] - sNW0RH49OP9j

**Comment by mf-code** [comment]
[5810] - jnjAwerN0JnG

**Comment by mf-code** [comment]
[5810] - b93MAXOTbR1F

**Comment by mf-code** [comment]
[5810] - KiE898KyyChI

**Comment by mf-code** [comment]
[5810] - bO10mpUkyoUS

**Comment by mf-code** [comment]
[5810] - G7eQ7nqJGK20

**Comment by mf-code** [comment]
[5810] - az4A7J8FsJIe

**Comment by mf-code** [comment]
[5810] - fsqhIIMrAmW6

**Comment by mf-code** [comment]
[5810] - XLggAYVafFXr

**Comment by mf-code** [comment]
[5810] - C3BUHDwPMMDd

**Comment by mf-code** [comment]
[5810] - cKbR8sWhlMDe

**Comment by mf-code** [comment]
[5810] - 7hDvriJyKM6z

**Comment by mf-code** [comment]
[5810] - openjdk8 - Allure report on Netlify: https://5f10113658f63276f18761a1--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5810] - openjdk10 - Allure report on Netlify: https://5f10122f4fd32957de1a81ba--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5812] - ZH4C3QUWKiKs

**Comment by mf-code** [comment]
[5812] - hFv6g1mCdf9z

**Comment by mf-code** [comment]
[5812] - ntQwC7xpWW6r

**Comment by mf-code** [comment]
[5812] - AmXedjCaGoVy

**Comment by mf-code** [comment]
[5812] - 4kcDTjY33sL2

**Comment by mf-code** [comment]
[5812] - UJJHOk2IfJZS

**Comment by mf-code** [comment]
[5812] - jFpWBidQxe7P

**Comment by mf-code** [comment]
[5812] - Ny6zGorEgdZK

**Comment by mf-code** [comment]
[5812] - oCu5mUoV1pbh

**Comment by mf-code** [comment]
[5812] - qmgZqKRqLBtY

**Comment by mf-code** [comment]
[5812] - cZ9OsD8thPCD

**Comment by mf-code** [comment]
[5812] - L1QNgKfnxprR

**Comment by mf-code** [comment]
[5812] - QWh5dNEMnQ4X

**Comment by mf-code** [comment]
[5812] - FUlbKsTCybzp

**Comment by mf-code** [comment]
[5812] - openjdk8 - Allure report on Netlify: https://5f1030e07a966d24913d05ec--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5812] - openjdk10 - Allure report on Netlify: https://5f1031d058f6323fcc87611e--musing-swartz-6fc060.netlify.app

**Pull request merged**